voxblox/include/voxblox/core/voxel.h

```diff
@@ -32,21 +28,37 @@ struct EsdfVoxel {
   // Relative direction toward parent. If itself, then either uninitialized
   // or in the fixed frontier.
   Eigen::Vector3i parent = Eigen::Vector3i::Zero();
-};
 
-struct OccupancyVoxel {
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
 
+struct OccupancyVoxel {
   float probability_log = 0.0f;
   bool observed = false;
 };
 
+struct SkeletonVoxel {
+  float distance = 0.0f;
+  uint8_t num_basis_points = 0u;
+  bool is_face = false;
+  bool is_edge = false;
+  bool is_vertex = false;
+  int64_t vertex_id = -1;
+};
+
+struct ThermalVoxel {
```

Comment:
I agree, could be useful for laser too. The observations field also seems tailored to your particular application, it can only store a very low number of observations, which I assume corresponds to some threshold you use (> 0?). So IMO either replace it with a bool, turn it into a weight or drop it (in which case you can set the intensity value to an invalid value to represent unobserved voxels)?
@helenol  @ZacharyTaylor wdyt?

Reply:
Sounds good, I'll switch to IntensityVoxel and add serialization.